core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractReactorStreamProcessingStrategy.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.strategy;
+
+import static org.mule.runtime.core.api.processor.ReactiveProcessor.ProcessingType.CPU_LITE_ASYNC;
+import static reactor.core.publisher.Flux.from;
+import static reactor.core.scheduler.Schedulers.fromExecutorService;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Startable;
+import org.mule.runtime.api.lifecycle.Stoppable;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.core.api.processor.ReactiveProcessor;
+import org.mule.runtime.core.internal.processor.strategy.AbstractStreamProcessingStrategyFactory.AbstractStreamProcessingStrategy;
+
+import java.util.function.Supplier;
+
+abstract class AbstractReactorStreamProcessingStrategy extends AbstractStreamProcessingStrategy implements Startable, Stoppable {
+
+  private final Supplier<Scheduler> cpuLightSchedulerSupplier;
+  private Scheduler cpuLightScheduler;
+  private final int parallelism;
+
+  AbstractReactorStreamProcessingStrategy(int subscribers,
+                                          Supplier<Scheduler> cpuLightSchedulerSupplier, int parallelism,
+                                          int maxConcurrency, boolean maxConcurrencyEagerCheck) {
+    super(subscribers, maxConcurrency, maxConcurrencyEagerCheck);
+    this.cpuLightSchedulerSupplier = cpuLightSchedulerSupplier;
+    this.parallelism = parallelism;
+  }
+
+  @Override
+  public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
+    reactor.core.scheduler.Scheduler cpuLightScheduler = fromExecutorService(decorateScheduler(getCpuLightScheduler()));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename "cpuLightScheduler" which hides the field declared at line 25. ([squid:HiddenFieldCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7708&issues=AWkQk2v5otIXnmsUMK6T&open=AWkQk2v5otIXnmsUMK6T)

[ISSUE_KEY]: AWkQk2v5otIXnmsUMK6T